### ISSUE TITLE
detect ERC20 transfers

### ISSUE THREAD

**Issue by Vasiliy-Bondarenko** [TrueBlocks/trueblocks-core#2466]
I am wondering how can you reliably detect ERC20 transfers if function ID overlaps with other function IDs. Here is the proof: https://www.4byte.directory/signatures/?bytes4_signature=0xa9059cbb
Can you clarify, please?

**Comment by tjayrush** [comment]
We can't. It's not possible.

But...there's a saving grace...the purpose of our tool is to extract the transaction necessary to do 18-decimal place accurate accounting on every transaction that an address is involved with.

I we encounter such as case (where the four-byte or event topic is a collision), we will pull that transaction as we would any other, but -- when we go to query that smart contract for `balances` (or `symbol` or `name`) we will get either garbeled results or much more likely no result at all. In that case, the "mistaken ERC20 transfer" will have zero value and, while it may appear in the list of transactions, it will have no effect on the accounting.

In other words, regular old-fashioned, off-chain double-entry accounting will come to the rescue (which is part of our design, since, as you point out, it's not possible for it to be perfect just from the on-chain data).

Does that answer your question? If yes, please let me know. I'll close this issue and copy it to our FAQ (You're not the first person to ask).

**Comment by Vasiliy-Bondarenko** [comment]
ok. got it. seems fine. thank you.

**Comment by tjayrush** [comment]
This has been added to the FAQ. Closing...
